lib/ood_core/job/adapters/sge/batch.rb

```diff
@@ -0,0 +1,194 @@
+# Object used for simplified communication with a SGE batch server
+#
+# @api private
+class OodCore::Job::Adapters::Sge::Batch
+  using OodCore::Refinements::HashExtensions
+  require "ood_core/job/adapters/sge/qstat_xml_j_r_listener"
+  require "ood_core/job/adapters/sge/qstat_xml_r_listener"
+  require "ood_core/job/adapters/sge/helper"
+  require "ood_core/refinements/hash_extensions"
+  require "tempfile"
+  require 'time'
+
+  begin
+    require "ood_core/job/adapters/drmaa"
+    require 'singleton'
+
+    DRMMA_TO_OOD_STATE_MAP = {
+      DRMAA::STATE_UNDETERMINED          => :undetermined,
+      DRMAA::STATE_QUEUED_ACTIVE         => :queued,
+      DRMAA::STATE_SYSTEM_ON_HOLD        => :queued_held,
+      DRMAA::STATE_USER_ON_HOLD          => :queued_held,
+      DRMAA::STATE_USER_SYSTEM_ON_HOLD   => :queued_held,
+      DRMAA::STATE_RUNNING               => :running,
+      DRMAA::STATE_SYSTEM_SUSPENDED      => :suspended,
+      DRMAA::STATE_USER_SUSPENDED        => :suspended,
+      DRMAA::STATE_USER_SYSTEM_SUSPENDED => :suspended,
+      DRMAA::STATE_DONE                  => :completed,
+      DRMAA::STATE_FAILED                => :completed
+    }
+
+    # The one and only connection with DRMAA
+    # Attempting to instantiate a DRMAA::Session more than once causes it to crash
+    class SessionSingleton < DRMAA::Session
+      include Singleton
+    end
+  rescue LoadError; end
+
+  class Error < StandardError; end
+
+  # @param opts [#to_h] the options defining this adapter
+  # @option opts [Batch] :batch The Sge batch object
+  #
+  # @api private
+  # @see Factory.build_sge
+  def initialize(config)
+    @cluster          = config.fetch(:cluster, nil)
+    @conf             = Pathname.new(config.fetch(:conf, nil))
+    @begin            = Pathname.new(config.fetch(:bin, nil))
```

Comment:
Change to bin

Reply:
Since you are considering fixing the tests to mock Open3 maybe verify or update tests to catch this error before fixing it.